src/realm/object-store/sync/sync_session.cpp

```diff
@@ -326,24 +327,29 @@ SyncSession::handle_refresh(const std::shared_ptr<SyncSession>& session, bool re
 
 SyncSession::SyncSession(SyncClient& client, std::shared_ptr<DB> db, const RealmConfig& config,
                          SyncManager* sync_manager)
-    : m_config(config)
-    , m_db(std::move(db))
-    , m_flx_subscription_store([this](bool use_flx_sync) -> std::shared_ptr<sync::SubscriptionStore> {
-        if (!use_flx_sync) {
+    : m_config{config}
+    , m_db{std::move(db)}
+    , m_flx_subscription_store{}
+    , m_original_sync_config{m_config.sync_config}
+    , m_migration_store{[&](bool is_flexible_sync) -> std::shared_ptr<sync::MigrationStore> {
+        using MigrationState = sync::MigrationStore::MigrationState;
+        if (is_flexible_sync) {
+            // Make sure the migration info is cleared
+            auto migration_store = sync::MigrationStore::create(m_db, [](MigrationState) {});
```

Comment:
it is a bit odd that we need to create an object just to clear the underlying data. was there a problem to have a static function?

Reply:
I have been thinking about this a little more, and I think there are cases when we have flx sessions where we don't want to clear the state (e.g, when performing a client reset in download_fresh_realm). We could of course overcome this by having a (config) parameter.

Reply:
I updated the instances where a new SyncSession is being generated to use the original_sync_config to prevent inadvertently clearing the migration state.

Reply:
I am not sure that's going to work. For pbs->flx we want to connect as flx so we can download the schema and do the initial bootstrap. If we connect as pbs we end up with a pbs realm.

Reply:
For `download_fresh_realm()`, it creates a new SyncSession which will fix up the configuration to FLX (if migrated) - I didn't wanted to use the updated sync_config in the new SyncSession, since it will think the app is natively using FLX.
For `make_client_reset_config()` in `create_sync_session()`, I also wanted to use the original sync_config, since it gets a SharedRealm in the `notify_after_client_reset` callback that would not point to the correct instance, and possibly reset the migrated state, if the updated sync_config was used.

Reply:
got it. That will work too. I forgot about the config being updated if needed when the SyncSession is created.